identity-enabler/deviceId-mobile-app/src/pages/CreatePresentation.svelte

```diff
@@ -39,6 +40,9 @@
     onMount(async () => {
         loadingScreen.set("Generating DataMatrix...");
         try {
+            // Fixes a bug where loading screen does not appear after the first time this
+            // page gets mounted.
+            await wait(0);
```

Comment:
can this be done using other approaches, for instance through a custom event raised right after the page gets mounted?

Reply:
In theory the `onMount` hook is supposed to fire right after the page gets mounted - from what I can tell, it does. I don't truly understand why the loading screen doesn't appear on subsequent mounts, because the loading screen component definitely mounts immediately and it shouldn't be CSS related if it displays correctly on first mount (and in other places it gets used). Since setting a 0-second timeout appears to solve the problem, it must be related to single-threaded nature of JS, but I'm not sure how or what is the cause.